autonomy/chain/mint.py

```diff
@@ -107,13 +111,29 @@ def mint_service(  # pylint: disable=too-many-arguments
     metadata_hash: str,
     chain_type: ChainType,
     agent_ids: List[int],
-    number_of_slots_per_agents: List[int],
+    number_of_slots_per_agent: List[int],
     cost_of_bond_per_agent: List[int],
     threshold: int,
 ) -> Optional[int]:
     """Publish component on-chain."""
+
+    if len(number_of_slots_per_agent) == 0:
```

Comment:
You should already have a check that raises an InvalindMintParameter when `if ( len(agent_ids) ==0)`
So, since in lines 126-130, you already checked that all the lengths are equal, lines 121 and 124 are not necessary. What you should check is instead that cost_of_bond_per_agent[i] != 0 &  number_of_slots_per_agent[i] != 0 for all i

Reply:
Addressed [293ff53](https://github.com/valory-xyz/open-autonomy/pull/1733/commits/293ff53a5f9ad6571e3cc0f6aa8d4cdb554a6521)